File: tests/ReturnOnly.pas
namespace Test;
interface

type
  Foo = public class
    function Bar : Integer;
    function TipoFonte: String;
  end;

implementation

function Foo.Bar : Integer;
begin
  result := 1;
end;

function Foo.TipoFonte: String;
begin
  if fonteAux.tipo.trim = '0' then
    result := 'P' //Fonte PUC
  else
    if fonteAux.tipo.trim = '1' then
      result := 'C' //Fonte Convênio
    else
      result := 'J'; //Fonte Projeto
end;

end.

File: tests/ReturnOnly.cs
namespace Test {
    public partial class Foo {
        public int Bar() {
            int result;
            result = 1;
            return result;
        }
        public string TipoFonte() {
            string result;
            if (fonteAux.tipo.trim == "0") result = "P"; /* Fonte PUC */
            else if (fonteAux.tipo.trim == "1") result = "C"; /* Fonte Convenio */
            else result = "J"; /* Fonte Projeto */
            return result;
        }
    }
}
